x-pack/plugins/ml/public/application/datavisualizer/data_drift/index_patterns_picker.tsx

```diff
@@ -0,0 +1,219 @@
+    navigateToPath(
+      `${nextStepPath}?${type === 'index-pattern' ? 'index' : 'savedSearchId'}=${encodeURIComponent(
+        id
+      )}`
+    );
+  };
+
+  const canEditDataView = dataViewEditor?.userPermissions.editDataView();
+
+  return (
+    <div data-test-subj="mlPageSourceSelection">
+      <EuiPageBody restrictWidth={1200}>
+        <>
+          <MlPageHeader>
+            <EuiFlexGroup>
+              <EuiFlexItem grow={true}>
+                <FormattedMessage
+                  id="xpack.ml.newJob.wizard.selectDataViewOrSavedSearch"
+                  defaultMessage="Select data view or saved search"
+                />
+              </EuiFlexItem>
+            </EuiFlexGroup>
+          </MlPageHeader>
+          <EuiPanel hasShadow={false} hasBorder>
+            <SavedObjectFinder
+              key="searchSavedObjectFinder"
+              onChoose={onObjectSelection}
+              showFilter
+              noItemsMessage={i18n.translate(
+                'xpack.ml.newJob.wizard.searchSelection.notFoundLabel',
+                {
+                  defaultMessage: 'No matching data views or saved searches found.',
+                }
+              )}
+              savedObjectMetaData={[
+                {
+                  type: 'search',
+                  getIconForSavedObject: () => 'search',
+                  name: i18n.translate(
+                    'xpack.ml.newJob.wizard.searchSelection.savedObjectType.search',
+                    {
+                      defaultMessage: 'Saved search',
+                    }
+                  ),
+                },
+                {
+                  type: 'index-pattern',
+                  getIconForSavedObject: () => 'indexPatternApp',
+                  name: i18n.translate(
+                    'xpack.ml.newJob.wizard.searchSelection.savedObjectType.dataView',
+                    {
+                      defaultMessage: 'Data view',
+                    }
+                  ),
+                },
+              ]}
+              fixedPageSize={20}
+              services={{
+                contentClient: contentManagement.client,
+                uiSettings,
+              }}
+            >
+              <EuiButton
+                size="m"
+                fill
+                iconType="plusInCircleFilled"
+                onClick={() => navigateToPath(createPath(ML_PAGES.DATA_DRIFT_CUSTOM))}
+                disabled={!canEditDataView}
+              >
+                <FormattedMessage
+                  id="xpack.ml.dataDrift.createDataViewButton"
+                  defaultMessage="Create a data view"
+                />
+              </EuiButton>
+            </SavedObjectFinder>
+          </EuiPanel>
+        </>
+      </EuiPageBody>
+    </div>
+  );
+};
+
+export const DataDriftIndexPatternsPicker: FC = () => {
+  const { sourceIp, destIp } = parse(location.search, {
+    sort: false,
+  }) as { sourceIp: string; destIp: string };
```

Comment:
nit: I find the naming a bit confusing, shall we use `sourceIndexPattern` instead?

Reply:
For consistency, I ended up changing it to reference and comparison here.